back/scripts/datasets/sirene.py

```diff
@@ -45,13 +46,110 @@ def __init__(self, config: dict):
 
     def _fetch_zip(self):
         if self.zip_filename.exists():
             return
         urllib.request.urlretrieve(self._config["url"], self.zip_filename)
 
+    def _fetch_xls_files(self):
+        xls_links = self._config.get("xls_urls_naf", [])
+        for file_url in xls_links:
+            file_name = file_url.split("/")[-1]
+            file_path = self.data_folder / file_name
+            if not file_path.exists():
+                urllib.request.urlretrieve(file_url, file_path)
+
+        xls_url_cat_ju = self._config.get("xls_url_cat_ju")
+
+        if xls_url_cat_ju:
+            file_name = xls_url_cat_ju.split("/")[-1]
+            file_path = self.data_folder / file_name
+            print(xls_url_cat_ju, file_path, file_path.exists())
+            if not file_path.exists():
+                print(file_path)
+                urllib.request.urlretrieve(xls_url_cat_ju, file_path)
+
+    def join_naf_level(
+        self,
+        base_df,
+        level,
+        nomenclature_filter_col="nomenclature_naf",
+        nomenclature_value="NAFRev2",
+    ):
+        """
+        Effectue la jointure avec le fichier correspondant au niveau (n1, n2, n3, n4, n5) sur base_df.
+        """
+        naf_file_path = self.data_folder / f"naf2008_liste_n{level}.xls"
+        naf_df = pd.read_excel(naf_file_path, header=2)
+        naf_df["Code"] = naf_df["Code"].astype(str).str.replace(".", "", regex=False)
+        naf_polars = pl.from_pandas(naf_df[["Code", "Libellé"]])
+
+        if level == 5:
+            slice_func = pl.col("naf8")
+        elif level == 1:
+            slice_func = pl.col("naf8").str.slice(-1)
+        elif level == 2:
+            slice_func = pl.col("naf8").str.slice(0, 2)
+        elif level == 3:
+            slice_func = pl.col("naf8").str.slice(0, 3)
+        elif level == 4:
+            slice_func = pl.col("naf8").str.slice(0, 4)
+
+        column_name = f"naf8_prefix_{level}"
+        base_df = base_df.with_columns(
+            pl.when(pl.col("nomenclature_naf") == "NAFRev2")
+            .then(slice_func)
+            .otherwise(None)
+            .alias(column_name)
+        )
+
+        merged_df = base_df.join(
+            naf_polars.rename({"Libellé": f"Libellé_naf_n{level}"}),
+            left_on=column_name,
+            right_on="Code",
+            how="left",
+        ).drop(column_name)
+        return merged_df
+
+    def join_juridical_level(
+        self, base_df, level, code_ju_col="code_ju", categories_ju_data=None
+    ):
+        """
+        Effectue la jointure avec les données juridiques correspondant au niveau (niv1, niv2, niv3) sur base_df.
+        """
+        base_df = base_df.with_columns(pl.col(code_ju_col).cast(pl.Utf8))
+        if level == 1:
+            slice_func = pl.col(code_ju_col).str.slice(0, 1)
+        elif level == 2:
+            slice_func = pl.col(code_ju_col).str.slice(0, 2)
+        elif level == 3:
+            slice_func = pl.col(code_ju_col)
```

Comment:
```suggestion
        if level == 3:
            slice_func = pl.col(code_ju_col)
        else:
            slice_func = pl.col(code_ju_col).str.slice(0, level)
```

or, provided max length is 3, just 

```suggestion
        slice_func = pl.col(code_ju_col, level)
```